# 1. Importar librerías

In [27]:
#Librerías para manipulación de datos
import pandas as pd

#Librerías para visualización
import matplotlib.pyplot as plt
import seaborn as sns
from plotly.subplots import make_subplots
import plotly.graph_objects as go

#Importar módulo de funciones
import sys
sys.path.insert(0, 'utils')  # Añadir el directorio 'utils' al PATH
import funciones
from importlib import reload
reload(funciones)  # Recarga el módulo

<module 'funciones' from 'c:\\recursos_humanos\\utils\\funciones.py'>

# 2. Importar datos

In [19]:
data = funciones.get_data('data_train')

In [20]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4410 entries, 0 to 4409
Data columns (total 27 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   InfoDate                 4410 non-null   dbdate 
 1   EmployeeID               4410 non-null   Int64  
 2   Age                      4410 non-null   Int64  
 3   DistanceFromHome         4410 non-null   Int64  
 4   MonthlyIncome            4410 non-null   Int64  
 5   NumCompaniesWorked       4410 non-null   float64
 6   PercentSalaryHike        4410 non-null   Int64  
 7   StockOptionLevel         4410 non-null   Int64  
 8   TrainingTimesLastYear    4410 non-null   Int64  
 9   YearsAtCompany           4410 non-null   Int64  
 10  YearsSinceLastPromotion  4410 non-null   Int64  
 11  YearsWithCurrManager     4410 non-null   Int64  
 12  TotalWorkingYears        4410 non-null   float64
 13  EnvironmentSatisfaction  4410 non-null   float64
 14  JobSatisfaction         

# 3. Análisis de Variables Categóricas

### 3.1. ¿Cuál fue el porcentaje de renuncias durante el año 2015?

In [21]:
funciones.univariado_barras(data, 'Attrition', 'v')

Attrition  Frecuencia Absoluta Frecuencia Relativa
0        No                 3760            85.261 %
1        Sí                  650            14.739 %

Aproximadamente el 15% de los empleados se retiraron en el año 2015, ya sea porque fueron despedidos o renunciaron.

### 3.2. ¿Cuál es el departamento que presentó el porcentaje más alto de retiros durante el año 2015?

In [28]:
funciones.analisisBivariado(data,['Attrition','Department'],'v','group')

Tabla de contingencia:


Department  Human Resources  Research & Development     Sales
Attrition                                                    
No                 0.036702                0.655585  0.307713
Sí                 0.078462                0.643077  0.278462

De los empleados que se retiraron, el 64.31% pertenecen al departamento de Investigación y Desarrollo. Además, de los empleados de Recursos Humanos, la proporción de empleados que se retiraron es el doble que la proporción de los que no se retiraron. En términos generales, el departamento con mayor número de retiros es Investigación y Desarrollo, pero hay que revisar Recursos Humanos porque el porcentaje de representación de los que se retiran es mayor que los que se quedan.

### 3.3. ¿Los retiros están relacionados con la frecuencia con la que los empleados deben viajar?

In [36]:
funciones.analisisBivariado(data, ['Attrition', 'BusinessTravel'], 'v','relative')

Tabla de contingencia:


BusinessTravel  Non-Travel  Travel_Frequently  Travel_Rarely
Attrition                                                   
No                0.110372           0.169681       0.719947
Sí                0.053846           0.296923       0.649231

De los empleados que se retiraron en el 2015, hubo más presencia de empleados que debían viajar frecuentemente. Además, de los empleados que debían viajar frecuentemente, la proporción de empleados que no se retiraron menor a la proporción de empleados que sí se retiraron, por lo que esta variable puede ser importante para predecir el retiro de un empleado.

### 3.4. ¿Qué influencia tiene el nivel de educación en el porcentaje de retiros?

In [39]:
funciones.analisisBivariado(data, ['Attrition', 'Education'], 'v', 'relative')

Tabla de contingencia:


Education  Bachelor  Below College   College    Doctor    Master
Attrition                                                       
No         0.389362       0.117021  0.186968  0.033245  0.273404
Sí         0.387692       0.107692  0.220000  0.029231  0.255385

No hay una influencia clara del nivel de educación con los retiros.